### ISSUE TITLE
scale and scale! have redundant methods

### ISSUE THREAD

**Issue by jiahao** [JuliaLang/julia#13690]
The base library contains many different methods for `scale` and `scale!` which look like they are pre- and post-multiplicative variants, but are actually not. For example:

```
scale(x::Number, y::Vector) = x * y
scale(x::Vector, y::Number) = x * y
```

These methods are redundant when the product is commutative, and they are wrong when it is not (e.g. quaternions, which are clearly `<: Number`).

We should either

1. Delete half of these methods so that `scale`-ing only means either pre- or post-multiplication, or
2. Fix the existing methods to not assume commutativity, i.e. distinguish between pre- or post-multiplications.

**Comment by StefanKarpinski** [comment]
Fixing it to not assume commutativity seems best.

**Comment by sarvjeetsinghghotra** [comment]
made a pull request #14402.

**Comment by pkofod** [comment]
Fixed by #14425 ?

**Comment by ivarne** [comment]
Seems like it was.
